kotlin-plugin/src/main/java/com/intellij/rt/coverage/instrumentation/kotlin/KotlinUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intellij.rt.coverage.instrumentation.kotlin;
+
+import com.intellij.rt.coverage.instrumentation.Instrumenter;
+
+import java.util.List;
+
+public class KotlinUtils {
+  private static Object ourClassLabel = null;
+  private static boolean ourIsKotlinClass = false;
+
+  /**
+   * Calls of this method with the same annotations object use buffered response.
+   * Use the same annotations object for one class.
+   * Use different annotations objects for different classes.
+   */
+  private static boolean hasKotlinAnnotation(List<String> annotations) {
+    if (ourClassLabel == annotations) {
```

Comment:
This information can be used from several filters. Also I need this information before filter instanced.

Reply:
I believe it's ok to repeat field per filter. Where do you need this information before filters are instantiated?

Reply:
I can just make this:
```
public static boolean isKotlinClass(Instrumenter context) {
  return context.getAnnotations().contains("Lkotlin/Metadata;");
}
```

But it iterates over annotations for every method in class. Current code avoids this.

If I store a field per filter(and calculate it inside) it will cost the same - one iteration per filter.

I need this information in `Builder.isApplicable` and builder is a kind of singleton => fields will be common for all calls.

What is wrong with current solution?

Reply:
you cache per class information in a static field, it's extremely uncommon thing. 
Ok, I'll check the code in IDE to see how it can be improved

Reply:
Methods are visited one by one from the same class, then comes the next class, so I don't need information about previous classes. 
I can store identityHashCode instead of object to avoid memory leaks.